### PR TITLE
Setup CompressImagesFunction as a Docker container

### PR DESCRIPTION
**Pull request by dabutvin**
part of #341 
still need to figure out the scaling part of this, but the container is ready to go

published at https://hub.docker.com/r/dabutvin/imgbot-compressimages

### CODE REVIEW

File: CompressImagesFunction/CommitSignature.cs [hunk 1]
```diff
@@ -8,8 +8,9 @@ namespace CompressImagesFunction
 {
     public class CommitSignature
     {
-        public static string Sign(string commitMessage, Stream privateKeyStream, string password)
+        public static string Sign(string commitMessage, string privateKey, string password)
         {
+            var privateKeyStream = new MemoryStream(Encoding.ASCII.GetBytes(privateKey));
```
Comment (dlemstra) [inline CompressImagesFunction/CommitSignature.cs]: Why are you not disposing the streams in this class?

File: Common/KnownEnvironmentVariables.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+using System;
+
+namespace Common
+{
+    public static class KnownEnvironmentVariables
+    {
+        public static string TMP
```
Comment (dlemstra) [inline Common/KnownEnvironmentVariables.cs]: Might be "prettier" to use `=>` here.

File: Dockerfile.CompressImages [hunk 3]
```diff
@@ -0,0 +1,14 @@
+FROM mcr.microsoft.com/dotnet/core/sdk:2.2 AS dotnet
+
+COPY . /src/
+RUN mkdir -p /home/site/wwwroot
+RUN cd /src/CompressImagesFunction && dotnet publish CompressImagesFunction.csproj -c Release --output /home/site/wwwroot
+
+# Native Binaries
+RUN cd /src/CompressImagesFunction && cp bin/Release/netstandard2.0/bin/runtimes/linux-x64/native/libgit2-4aecb64.so /home/site/wwwroot/bin/
+RUN cd /src/CompressImagesFunction && cp bin/Release/netstandard2.0/bin/runtimes/linux-x64/native/Magick.NET-Q16-x64.Native.dll.so /home/site/wwwroot/bin/
```
Comment (dlemstra) [inline Dockerfile.CompressImages]: Maybe copy `*.so` here?

File: CompressImagesFunction/CommitSignature.cs (hunk 1 above)
Comment (dabutvin) [reply CompressImagesFunction/CommitSignature.cs]: I always forget about that 🙃

File: Dockerfile.CompressImages (hunk 3 above)
Comment (dabutvin) [reply Dockerfile.CompressImages]: copy every `*.so` file in the bin?
It would make upgrading libgit slightly easier, but it feels kind of reckless

**Pull request merged**